src/js/components/form-elements/v2/FileSelect.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useCallback } from 'react';
+
+import PropTypes from 'prop-types';
+import { useDropzone } from 'react-dropzone';
+
+import Button from 'components/form-elements/Button';
+import useTranslate from 'hooks/useTranslate';
+
+import './style.scss';
+
+const FileSelect = ({
+  height,
+  width,
+  dropzoneText,
+  buttonLabel,
+  className,
+  multiple,
+  maxFiles,
+  allowedExtensions,
+  ...fieldProps
+}) => {
+  const onDrop = useCallback((acceptedFiles) => {
+    fieldProps.onChange?.(multiple ? acceptedFiles : acceptedFiles[0]);
+  }, []);
+
+  const translate = useTranslate();
+
+  const getFileExtension = (file) => file.path.split('.')?.[1];
+
+  const validateFileType = (file) => {
+    if (!allowedExtensions.length || allowedExtensions.includes(getFileExtension(file))) {
+      return null;
+    }
+
+    return {
+      code: 'invalid-extension',
+      message: translate(
+        'react.default.error.invalidFileExtension.label',
+        `File extension should be one of: ${allowedExtensions.join(', ')}`,
+        [allowedExtensions.join(', ')],
+      ),
+    };
+  };
+
+  const {
+    getRootProps, getInputProps, open, acceptedFiles, fileRejections,
+  } = useDropzone({
+    onDrop,
+    noClick: true,
+    noKeyboard: true,
+    validator: validateFileType,
```

Comment:
but I guess this hook from the library expects to return a falsy value if the validation passes, hence my previous comment doesn't make sense, since we are not able to adjust that?

Reply:
This is not the validation for our libraries, I answered it in the comment above. I tested it with our libraries and it is working fine. In case someone has some problems with adjusting it, it can also be done by the workaround by passing the `setValue` function.